tests/js/set-tests.js

```diff
@@ -0,0 +1,392 @@
+            name: "Person",
+            properties: {
+                Names: "string<>"
+            }
+        };
+
+        const realm = new Realm({ schema: [people_schema] });
+        const schema = realm.schema;
+
+        //
+        // Set.has() functionality
+        //
+
+        realm.write(() => {
+            // put some names in our database
+            realm.create(people_schema.name, {
+                Names: ["Kenneth", "Andrew", "Steffen"]
+            });
+        });
+
+        let teams = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, teams.length, "There should be only one JS team");
+        let js_team = teams[0];
+        TestCase.assertEqual(3, js_team.Names.length, "There should be 3 people in the JS team");
+        TestCase.assertEqual(true, js_team.Names.has("Kenneth"), "Kenneth should be in the JS team");
+        TestCase.assertEqual(false, js_team.Names.has("Jason"), "Jason shouldn't be in the JS team");
+
+        // add one JS team member, delete another one
+        realm.write(() => {
+            js_team.Names.add("Kræn").delete("Kenneth");
+        });
+        teams = realm.objects(people_schema.name);
+        TestCase.assertEqual(1, teams.length, "There should be only one JS team");
+        js_team = teams[0];
+        TestCase.assertEqual(3, js_team.Names.length, "There should be 3 people in the JS team");
+        TestCase.assertEqual(false, js_team.Names.has("Kenneth"), "Kenneth shouldn't be in the JS team");
+
+        // create another team with two people
+        realm.write(() => {
+            realm.create(people_schema.name, {
+                Names: ["Christian", "Claus"]
+            });
+        });
+        teams = realm.objects(people_schema.name);
+        TestCase.assertEqual(2, teams.length, "There should be two teams");
+        js_team = teams[0];
+        let java_team = teams[1];
+        TestCase.assertEqual(3, js_team.Names.length, "There should be 3 people in the JS team");
+        TestCase.assertEqual(2, java_team.Names.length, "There should be 2 people in the Java team");
+
+        TestCase.assertEqual(false, java_team.Names.has("Kenneth"), "Kenneth shouldn't be in the Java team");
+        TestCase.assertEqual(true, java_team.Names.has("Claus"), "Claus should be in the Java team");
+
+        //
+        // Set.clear() functionality
+        //
+
+        // remove everyone from the JS team
+        realm.write(() => {
+            js_team.Names.clear();
+        });
+
+        teams = realm.objects(people_schema.name);
+        TestCase.assertEqual(2, teams.length, "There should be two teams");
+        js_team = teams[0];
+        java_team = teams[1];
+        TestCase.assertEqual(0, js_team.Names.length, "There should be no people in the JS team");
+        TestCase.assertEqual(2, java_team.Names.length, "There should be 2 people in the Java team");
+
+        realm.close();
+    },
+
+    // TODO:  Move Sync tests to separate file
```

Comment:
Why is that?

Reply:
I'm actually not sure.  It just seems to be the current convention that Sync tests don't reside in the same place as the other tests.
They seem to be guarded both in inclusion ([here](https://github.com/realm/realm-js/blob/5316468d40a6c5ab26825e1ac0fc289665029ffc/tests/js/index.js#L63)) and in the tests themselves by `if (!global.enableSyncTests) return;`.  I'm not sure if this convention has a deeper purpose.

Reply:
The convention comes from the time where we didn't run sync tests on RN (which is still true).

Reply:
Do we want to keep both these guards as standard practice?

Reply:
@kneth @kraenhansen  I moved the Sync tests to a separate file.  Please resolve if that's a suitable solution.